tests/unit/s2n_ecc_evp_test.c

```diff
@@ -345,5 +360,119 @@ int main(int argc, char **argv) {
+        s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(conn);
+        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "20190802"));
+        EXPECT_SUCCESS(s2n_connection_get_security_policy(conn, &security_policy));
+
+        /* Ensure that the applied security policy uses the correct ecc preferences */
+        const uint8_t EXPECTED_ECC_PREF_COUNT = 2;
+        EXPECT_EQUAL(security_policy->ecc_preferences, &s2n_ecc_preferences_20140601);
+        EXPECT_EQUAL(security_policy->ecc_preferences->count, EXPECTED_ECC_PREF_COUNT);
+
+        /* Setup & verify invalid curves, which will be selected by a malicious server */
+        const struct s2n_ecc_named_curve* const unrequested_curves[] = {
+                &s2n_unsupported_curve,
+                &s2n_ecc_curve_secp521r1,
+        };
+
+        /* Test that client's preferred ecc's do NOT include the curves selected by the server */
+        for (size_t ix = 0; ix < EXPECTED_ECC_PREF_COUNT; ix++) {
+            const struct s2n_ecc_named_curve* curve = security_policy->ecc_preferences->ecc_curves[ix];
+            for (uint8_t iy = 0; iy < s2n_array_len(unrequested_curves); iy++) {
+                EXPECT_NOT_EQUAL(curve->name, unrequested_curves[iy]->name);
+            }
+        }
+
+        /* Verify that the client broadcasts an error code when the server attempts to
+         * negotiate a curve that was never offered */
+        for (size_t i = 0; i < s2n_array_len(unrequested_curves); i++) {
+            struct s2n_ecc_evp_params server_params = {0}, client_params = {0};
+            struct s2n_stuffer wire;
+            struct s2n_blob ecdh_params_sent, ecdh_params_received;
+
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire, 1024));
+
+            /* Server maliciously chooses an unsupported curve */
+            server_params.negotiated_curve = unrequested_curves[i];
+            EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&server_params));
+            EXPECT_NOT_NULL(server_params.evp_pkey);
+            /* Server sends the public */
+            EXPECT_SUCCESS(s2n_ecc_evp_write_params(&server_params, &wire, &ecdh_params_sent));
+            /* Client reads the public */
+            struct s2n_ecdhe_raw_server_params ecdhe_data = {0};
+            EXPECT_SUCCESS(s2n_ecc_evp_read_params(&wire, &ecdh_params_received, &ecdhe_data));
+            EXPECT_FAILURE_WITH_ERRNO(
+                    s2n_ecc_evp_parse_params(conn, &ecdhe_data, &client_params), S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
+
+            /* The client didn't agree on a curve*/
+            EXPECT_NULL(client_params.negotiated_curve);
+
+            /* Clean up */
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire));
+            EXPECT_SUCCESS(s2n_ecc_evp_params_free(&server_params));
+            EXPECT_SUCCESS(s2n_ecc_evp_params_free(&client_params));
+        }
+    }
+
+    /* Batch test that the client selects a curve that was initially offered in EC preferences
```

Comment:
I'm not convinced this test case is necessary, we already test the happy path case in [here](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_ecc_evp_test.c#L329). Is there something that you think this test is missing?

Reply:
Ok, removed